torchvision/csrc/io/image/cuda/encode_decode_jpegs_cuda.h

```diff
@@ -2,14 +2,15 @@
 
 #include <torch/types.h>
 #include "../image_read_mode.h"
+#include "decode_jpegs_cuda.h"
 #include "encode_jpegs_cuda.h"
 
 namespace vision {
 namespace image {
 
-C10_EXPORT torch::Tensor decode_jpeg_cuda(
-    const torch::Tensor& data,
-    ImageReadMode mode,
+C10_EXPORT std::vector<torch::Tensor> decode_jpegs_cuda(
```

Comment:
Add a comment with example usage and comment about thread safety. Can multiple threads call this function concurrently?

Reply:
Also, what happens on errors? If a single image fails, will it fail the whole batch?

Does it decode everything in parallel?

Reply:
The C++ functions are primarily meant to be used by the PyBind code only. I didn't really intend for them to be used by libtorch users, but I suppose there isn't really any harm in it either. Added docstrings.

Reply:
To summarize: 
- Yes, both functions are thread-safe.
- Errors in one image fails the whole function. No partial results are returned.
- The decoding happens in parallel